src/map.cpp

```diff
@@ -8175,22 +8169,25 @@ template<typename Functor>
     }
 }
 
-void map::scent_blockers( bool (&blocks_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
-                          bool (&reduces_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
+void map::scent_blockers( std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &blocks_scent,
+                          std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &reduces_scent,
```

Comment:
> What does this do in the  map  class? 

Dunno. It was that way when I looked at it.

Instead of moving it directly into `scent_map`, I would move its content into the map cache system. It depends only on the terrain / furniture / vehicles and we already cache various data derived from them (transparency, light, outside).